src/coreclr/vm/exceptionhandling.cpp

```diff
@@ -4537,7 +4537,7 @@ VOID UnwindManagedExceptionPass2(PAL_SEHException& ex, CONTEXT* unwindStartConte
     EEPOLICY_HANDLE_FATAL_ERROR(COR_E_EXECUTIONENGINE);
 }
 
-extern void* g_hostingApiFrameAddress;
+extern void* g_hostingApiReturnAddress;
```

Comment:
Is it possible for two different threads to call the hosting API in parallel?

Reply:
I don't think that calling coreclr_initialize / coreclr_execute_assembly in parallel from multiple threads would work. I guess it would likely crash somewhere.

Reply:
Btw, the original version of my previous change was putting the hosting apis into a separate named section. Then I could use linker generated symbols for the section start / end and compare the PC to that without saving any return address etc. However, the section names format and way to get section range is different for macOS / Linux and I wasn't sure about the other Unixes. That made me move to this solution that should work everywhere.